modules/rte_mis_state/rte_mis_state.module

```diff
@@ -757,101 +760,85 @@ function rte_mis_state_user_presave(UserInterface $account) {
-        if (!$term_details) {
-          \Drupal::messenger()->addError(t('School Related to your account is not found.'));
-          return;
-        }
-        $user = \Drupal::entityTypeManager()->getStorage('user')->loadByProperties([
-          'name' => $term_details->getName(),
-        ]);
-        if ($user) {
-          $user = reset($user);
-          $user_mobile_number = $user->get('field_phone_number')->value;
-          $user_email_id = $user->getEmail();
-
-          // Check if sending email is enabled.
-          $email_status = $mailConfig->get('school_notification_email.enabled_email_notification') ?? '';
-          // Email subject & body.
-          $email_subject = $mailConfig->get('school_notification_email.email_notification_subject') ?? '';
-          $raw_email_body = $mailConfig->get('school_notification_email.email_notification_message') ?? '';
-
-          // Replace the tokens.
-          $modified_email_body = str_replace([
-            '!user',
-            '!existing_state',
-            '!modified_state',
-          ],
-          [
-            $term_details->getName(),
-            $original_sid,
-            $current_sid,
-          ], $raw_email_body);
-
-          if ($email_status) {
-            $params = [];
-            if (!empty($email_subject) && !empty($raw_email_body)) {
-              $params['subject'] = $email_subject;
-              $params['message'] = $modified_email_body;
-            }
-
-            $email_sent = \Drupal::service('plugin.manager.mail')->mail('rte_mis_state', 'state_change', $user_email_id, \Drupal::languageManager()->getCurrentLanguage(), $params);
-          }
-
-          $sms_status = $mailConfig->get('mobile_number_notification.enable_mobile_number_notification') ?? '';
-          $sms_body = $mailConfig->get('mobile_number_notification.mobile_number_notification_message') ?? '';
-
-          $modified_sms_body = str_replace([
-            '!user',
-            '!existing_state',
-            '!modified_state',
-          ],
-          [
-            $term_details->getName(),
-            $original_sid,
-            $current_sid,
-          ], $sms_body);
-
-          $sms_sent = $sms_status ? mobile_number_send_sms($user_mobile_number, $modified_sms_body) : NULL;
+    // Get the entity id of the linked school.
+    $school_entity_id = $transition->entity_id->getString();
+    // Now load user based on the identified mini node.
+    $user = \Drupal::entityTypeManager()->getStorage('user')->loadByProperties([
+      'field_school_details' => $school_entity_id,
+    ]);
+    $sms_provider = \Drupal::service('sms.provider');
+
+    if ($user) {
+      $user = reset($user);
+      $user_mobile_number = $user->get('field_phone_number')->value;
+      $user_email_id = $user->getEmail();
+
+      // Check if sending email is enabled.
+      $email_status = $mailConfig->get('school_notification_email.enabled_email_notification') ?? '';
```

Comment:
Can we check the email status first, and then do all the logic/processing? This can be useful if email notification is not enabled then, why do so much processing?
This applies to SMS as well.

Reply:
sure. That will be better.